messages/teleporter/message_manager.go

```diff
@@ -323,3 +294,40 @@ func (m *messageManager) parseTeleporterMessage(warpMessageID ids.ID, warpPayloa
 	}
 	return teleporterMessage, nil
 }
+
+func (m *messageManager) getTeleporterMessenger(destinationClient vms.DestinationClient) (*teleportermessenger.TeleporterMessenger, error) {
+	client, ok := destinationClient.Client().(ethclient.Client)
+	if !ok {
+		m.logger.Error(
+			"Destination client is not an Ethereum client.",
+			zap.String("destinationBlockchainID", destinationClient.DestinationBlockchainID().String()),
+		)
+		return nil, errors.New("destination client is not an Ethereum client")
+	}
+
+	// Get the teleporter messenger contract
+	teleporterMessenger, err := teleportermessenger.NewTeleporterMessenger(common.BytesToAddress(m.protocolAddress[:]), client)
+	if err != nil {
+		m.logger.Error(
+			"Failed to get teleporter messenger contract",
+			zap.Error(err),
+		)
+		return nil, err
+	}
```

Comment:
I think we should actually just panic in both of these error cases. They're both "impossible" cases. `teleportermessenger.NewTeleporterMessenger` doesn't actually call anything on the client, it just shoves it into a struct that can be used to call functions on the contract.

Right now, these errors will get bubbled up and logged and swallowed, but if either of these two calls ever fail, every call to them will fail every time.

Reply:
that makes sense, updated.

Reply:
Great! You can also remove the `error` from the return types of the function now that it never returns an error